src/util/amount.rs

```diff
@@ -866,18 +866,28 @@ pub mod serde {
     //! #[derive(Serialize, Deserialize)]
     //! pub struct HasAmount {
     //!     #[serde(with = "bitcoin::util::amount::serde::as_btc")]
-    //!     pub amount: Amount,
+    //!     pub btc_float_amount: Amount,
+    //!     #[serde(with = "bitcoin::util::amount::serde::as_sat")]
+    //!     pub int_sat_amount: Amount,
+    //! }
+    //!
+    //! #[derive(Deserialize)]
+    //! pub struct HasUserInputtedAmount {
+    //!     #[serde(deserialize_with = "bitcoin::util::amount::parse")]
+    //!     pub input_amount: Amount,
     //! }
     //! ```
 
+    use std::str::FromStr;
+
     use serde::{Deserialize, Deserializer, Serialize, Serializer};
     use util::amount::{Amount, Denomination, SignedAmount};
 
     /// This trait is used only to avoid code duplication and naming collisions
     /// of the different serde serialization crates.
     ///
     /// TODO: Add the private::Sealed bound in next breaking release
```

Comment:
Probably we need here
```suggestion
    #[doc = hidden]
    // TODO: Add the private::Sealed bound in next breaking release
```

Reply:
made it into a regular comment